src/lib/core/components/BoxStatsTable.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useMemo } from 'react';
+import { PromiseHookState } from '../hooks/promise';
+import { StudyEntity, Variable } from '../types/study';
+import { BoxplotDataWithCoverage } from './visualizations/implementations/BoxplotVisualization';
+import { BoxplotStatsTable } from '../api/DataClient/types';
+import { fixVarIdLabel } from '../utils/visualization';
+
+export interface Props {
+  data: PromiseHookState<BoxplotDataWithCoverage | undefined>;
+  entities?: StudyEntity[];
+  descriptor?: { type: string; configuration: unknown };
+  xAxisVariable?: Variable;
+  overlayVariable?: Variable;
+  facetVariable?: Variable;
+  independentAxisLabel?: string;
+}
+
+export function BoxStatsTable({
+  data,
+  entities,
+  descriptor,
+  xAxisVariable,
+  overlayVariable,
+  facetVariable,
+  independentAxisLabel,
+}: Props) {
+  const collectionVariableDetails = useMemo(
+    () =>
+      data?.value?.computedVariableMetadata?.collectionVariable
+        ?.collectionVariableDetails,
+    [data]
+  );
+
+  return (
+    <div
+      className={'BoxStatsTable'}
+      style={{ marginTop: 15, marginLeft: '0.75rem' }}
+    >
+      <table>
+        <tbody>
+          <tr>
+            <th>{'X-axis'}</th>
+            {facetVariable != null ? <th>{'Facet'}</th> : null}
+            <th>{'Parameter'}</th>
+            <th>{'Statistic'}</th>
+            <th>{'p-value'}</th>
+            <th>{'Method'}</th>
+          </tr>
+          {data.value?.statsTable?.map((data: BoxplotStatsTable) => (
+            <tr>
+              <td>
+                {descriptor?.type === 'abundance'
+                  ? overlayVariable == null
+                    ? independentAxisLabel
+                    : collectionVariableDetails && entities
+                    ? fixVarIdLabel(
+                        data.xVariableDetails?.value ?? '',
+                        collectionVariableDetails,
+                        entities
+                      )
+                    : data.xVariableDetails?.value
+                  : overlayVariable != null
```

Comment:
I'm not quite following this totally, it seems like we have
```
descriptor.type === 'abundance'
    ? overlayVar == null
        ? (...)
        : (...)
    : overlayVar != null
        ? (...)
        : (...)
```
where we are asking if overlayVar is/isn't null twice? What am i missing?

Reply:
@asizemore thank you for your reviews! 👍 
This part is made this morning to address bug for abundance where it is now displaying plot when selecting boxplot: because its X and Y are computed variables. So, the condition starts from abundance or not. so the first condition, ? overlayVar == null is when type is abundance case, and the second condition, : overlayVar != null is general case, just not abundance. The reason why I introduced the first condition, ? overlayVar == null, is to display X axis contents for the initial plot or facet variable is selected (while no overlayVariable)

Reply:
No problem! I was also confused about that when I made it 😄